language-plutus-core/src/Language/PlutusCore/Name.hs

```diff
@@ -23,22 +24,30 @@ import qualified Data.Map             as M
 import           Data.Text.Encoding   (decodeUtf8)
 import           PlutusPrelude
 
+import           Language.Haskell.TH.Syntax (Lift)
+-- for bytestring Lift instance
+import           Instances.TH.Lift          ()
+
 -- | A 'Name' represents variables/names in Plutus Core.
 data Name a = Name { nameAttribute :: a
                    , nameString    :: BSL.ByteString -- ^ The identifier name, for use in error messages.
                    , nameUnique    :: Unique -- ^ A 'Unique' assigned to the name during lexing, allowing for cheap comparisons in the compiler.
                    }
-            deriving (Functor, Show, Generic, NFData)
+            deriving (Functor, Show, Generic, NFData, Lift)
 
 -- | We use a @newtype@ to enforce separation between names used for types and
 -- those used for terms.
 newtype TyName a = TyName { unTyName :: Name a }
-    deriving Show
-    deriving newtype (Eq, Functor, NFData, Pretty, Debug)
+    deriving (Show, Lift)
+    deriving newtype (Eq, Ord, Functor, NFData, Pretty, Debug)
 
 instance Eq (Name a) where
     (==) = (==) `on` nameUnique
 
+-- derived instance is not okay since it uses the attribute!
+instance Ord (Name a) where
+  n1 <= n2 = (nameString n1) <= (nameString n2) && (nameUnique n1) <= (nameUnique n2)
```

Comment:
Is there a reason to not simply use `nameUnique` here? That seems like it would be faster.

Reply:
Ah - is `nameUnique` globally unique? For some reason I thought it was only unique modulo having the same name.